tests/dide/test_mounts.py

```diff
@@ -0,0 +1,144 @@
+from unittest import mock
+
+import pytest
+
+from hipercow.dide import mounts
+
+
+def test_can_parse_cifs_output():
+    m = mounts._parse_unix_mount_entry(
+        "//projects/other on /path/local type cifs (rw,relatime)"
+    )
+    assert m == mounts.Mount("projects", "other", "/path/local")
+
+
+def test_can_parse_mounts_on_unix(mocker):
+    data = b"""//projects/other on /path/local type cifs (rw,relatime)
+//projects/other2 on /path/local2 type cifs (rw,relatime)"""
+    response = mock.MagicMock(spec=CompletedProcess)
+    response.stdout = data
+    mocker.patch("subprocess.run", return_value=response)
+    res = mounts._detect_mounts_unix()
+    assert len(res) == 2
+    assert res[0] == mounts.Mount("projects", "other", "/path/local")
+    assert res[1] == mounts.Mount("projects", "other2", "/path/local2")
+
+
+def test_can_clean_dide_hostname():
+    assert mounts._clean_dide_hostname("qdrive") == "qdrive"
+    assert mounts._clean_dide_hostname("qdrive.dide.ic.ac.uk") == "qdrive"
+    assert mounts._clean_dide_hostname("qdrive.other") == "qdrive.other"
+
+
+def test_can_parse_mounts_on_windows(mocker):
+    data = b"""#TYPE Microsoft.Management.Infrastructure.CimInstance#ROOT/Microsoft/Windows/SMB/MSFT_SmbMapping
+"Status","LocalPath","RemotePath","RequireIntegrity","RequirePrivacy","UseWriteThrough","PSComputerName"
+"OK","I:","\\\\wpia-hn\\hipercow","False","False","False",
+"OK","Y:","\\\\wpia-hn2.hpc.dide.ic.ac.uk\\Climate","False","False","False",
+"Disconnected","Z:","\\\\wpia-hn\\all-wpia-hn","False","False","False","""  # noqa: E501
+    response = mock.MagicMock(spec=CompletedProcess)
+    response.stdout = data
+    mocker.patch("subprocess.run", return_value=response)
+    res = mounts._detect_mounts_windows()
+    assert len(res) == 2
+    assert res[0] == mounts.Mount("wpia-hn", "hipercow", "I:")
+    assert res[1] == mounts.Mount("wpia-hn2.hpc", "Climate", "Y:")
+
+
+def test_can_get_correct_smb_type():
+    assert mounts._unix_smb_mount_type("Linux") == "cifs"
+    assert mounts._unix_smb_mount_type("Darwin") == "smbfs"
+
+
+def test_throw_on_remap_with_no_mounts():
+    with pytest.raises(Exception, match="Can't map local directory"):
+        mounts.remap_path(Path.cwd(), [])
+
+
+def test_can_remap_path():
+    m = [mounts.Mount("host", Path("/hostmount"), Path("/local"))]
+    path = Path("/local/path/to/dir")
+    res = mounts.remap_path(path, m)
+    assert res == mounts.PathMap(
+        path, m[0], Path("V:"), relative=Path("path/to/dir")
+    )
+
+
+def test_throw_if_two_plausible_mounts():
+    m = [
+        mounts.Mount("host1", Path("/path1"), Path("/local/path")),
+        mounts.Mount("host2", Path("/path2"), Path("/local")),
+    ]
+    path = Path("/local/path/to/dir")
+    with pytest.raises(Exception, match="More than one plausible"):
+        mounts.remap_path(path, m)
+
+
+def test_preserve_drive_letter_if_given():
+    m = [mounts.Mount("host", Path("/hostmount"), Path("P:"))]
```

Comment:
```suggestion
    m = [mounts.Mount("host", Path("/hostmount"), Path("P:/"))]
```

Reply:
I've fixed this by stripping the trailing slash earlier, perhaps that's not ideal?